src/code/z_message.c

```diff
@@ -34,7 +177,33 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80149C18.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80149EBC.s")
+void Message_FindMessage(GlobalContext* globalCtx, u16 textId) {
+    const char* foundSegment;
+    const char* nextSegment;
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Font* font = &msgCtx->font;
+    MessageTableEntry* msgEntry = (MessageTableEntry*)msgCtx->messageEntryTable;
```

Comment:
```suggestion
    MessageTableEntry* msgEntry = msgCtx->messageEntryTable;
```
Cast seems unnecessary.